dvc/repo/artifacts.py

```diff
@@ -1,43 +1,57 @@
 import logging
-import os
 import re
-from typing import TYPE_CHECKING, Dict
+from pathlib import Path
+from typing import Dict, Optional
 
 from dvc.annotations import Artifact
-from dvc.dvcfile import FileMixin
+from dvc.dvcfile import PROJECT_FILE
+from dvc.exceptions import InvalidArgumentError
+from dvc.repo import Repo
 from dvc.utils import relpath
-
-if TYPE_CHECKING:
-    from dvc.repo import Repo
+from dvc.utils.serialize import modify_yaml
 
 logger = logging.getLogger(__name__)
 
 
-NAME_RE = re.compile(r"^[a-z]([a-z0-9-]*[a-z0-9])?$")
+# Constants are taken from GTO.
+# When we make it a dependency, we can import them instead
+SEPARATOR_IN_NAME = ":"
+DIRNAME = r"[a-z0-9-_./]+"
+NAME = r"[a-z0-9]([a-z0-9-/]*[a-z0-9])?"
+NAME_RE = re.compile(f"^{NAME}$")
+FULLNAME = f"((?P<dirname>{DIRNAME}){SEPARATOR_IN_NAME})?(?P<name>{NAME})"
+FULLNAME_RE = re.compile(f"^{FULLNAME}$")
 
 
 def name_is_compatible(name: str) -> bool:
     return bool(NAME_RE.search(name))
 
 
+wrong_name_message = (
+    "Can't use '%s' as artifact name (ID)."
+    " You can use letters and numbers, and use '-' as separator"
+    " (but not at the start or end)."
+)
+
+
 def check_name_format(name: str) -> None:
     if not name_is_compatible(name):
         logger.warning(
-            "Can't use '%s' as artifact name (ID)."
-            " You can use letters and numbers, and use '-' as separator"
-            " (but not at the start or end). The first character must be a letter.",
+            wrong_name_message,
             name,
         )
 
 
-class ArtifactsFile(FileMixin):
-    from dvc.schema import SINGLE_ARTIFACT_SCHEMA as SCHEMA
-
-    def dump(self, stage, **kwargs):
-        raise NotImplementedError
-
-    def merge(self, ancestor, other, allowed=None):
-        raise NotImplementedError
+def check_for_nested_dvc_repo(dvcfile: Path):
```

Comment:
What's the harm in allowing this?

Reply:
It's confusing, since if you add it like this, you later won't find it (it's out of scope for this DVC repo). Since Studio is the only user for this API, for now if the user adds the model like this (if we disable check), he won't find it in Studio MR. Makes sense to make this check on DVC side to me.

Reply:
As i have said before, dvc does not support nedted subrepos, so it does not make sense to add an error message about this.